src/main/java/org/jenkinsci/plugins/credentialsbinding/impl/SecretBuildWrapper.java

```diff
@@ -53,10 +63,13 @@
     }
 
     @Override public Environment setUp(AbstractBuild build, final Launcher launcher, BuildListener listener) throws IOException, InterruptedException {
-        final List<MultiBinding.MultiEnvironment> m = new ArrayList<MultiBinding.MultiEnvironment>();
-        for (MultiBinding binding : bindings) {
-            m.add(binding.bind(build, build.getWorkspace(), launcher, listener));
-        }
+        final List<MultiBinding.MultiEnvironment> m = Lists.transform(build.getActions(SharedMultiEnvAction.class),
+                new Function<SharedMultiEnvAction, MultiBinding.MultiEnvironment>() {
+                    public MultiBinding.MultiEnvironment apply (SharedMultiEnvAction action) {
+                        return action.get();
```

Comment:
:bug: There is no way this can work. You probably do not realize it because there are no tests for `FileBinding` or `ZipFileBinding` with `SecretBuildWrapper`.